### DISCUSSION TITLE
old Tine 2015 Installation update not possible, need next major Version x.2016.zip

### DISCUSSION THREAD

**Discussion by 0NDevil** [tine20/tine20#7405] [Q&A] [answered]
Hello,
my old vServer was killed by hoster because the server age.
Now i`m trying to update this old Tine (2015) version for my new Server but the oldest File i found is "tine20-allinone_2017.02.5.zip" and the update process stop!
**Error:** 
php setup.php --update
PHP Fatal error:  Declaration of Tinebase_Acl_Roles::getMultiple($_ids) must be compatible with Tinebase_Controller_Record_Abstract::getMultiple($_ids, $_ignoreACL = false) in /var/www/vhosts/domain.de/tine.domain.de/Tinebase/Acl/Roles.php on line 959
**Tine.log:**
Tinebase_Exception -> Major version jumps are not allowed. Upgrade your current major Version to the most recent minor Version, then upgrade to the most recent next major version. Repeat until you reached the desired major version you want to upgrade to.

So i need a 2016 Tine version (which should be the next major version)
If anyone have a old Version in his archiv ?

**Comment by lab-at-nohl** [comment, marked as answer]
I might help you. Do you still need support? I have documented [my upgrade path](https://www.nohl.eu/tine-20/legacy-migration-path/) (production/working) almost until the release cycle changed. For my path I have kept the releases, too. 

However, if you are familiar with the CLI (you are, I see), updating Tine means generally changing the Database according to the newly downloaded files. So you could download an older version (i.e. choose the corresponding git tag), set up the [build from git](https://github.com/tine20/tine20/wiki/EN%3Abuild-from-git) and do `php setup.php --update` for whatever release you like to... 

Edit: Corrected wiki page to the right one...

Until recently you need always the latest minor of the next major version. For 2016 two releases exist, 2016.03 and 2016.09. Thus, as an example if you are an an early 2016.03 version, go to 2016.**03.4** (latest minor) before upgrading to 2016.**09.7** (latest minor). Anyhow, in my log I noted that I had to upgrade to 2016.**09.6** before but that might be related to my personal setup. 

If you have no success with the git versions, write me an email. 

Best,
Johannes

**Comment by 0NDevil** [comment]
Hello Johannes,

Thank you for your detailed answer.
I had almost given up hope, but had been ill for the last few weeks and had other concerns, hence the late reply.

I checked again which version I have exactly and found this in the release notes:
Release: Coriander (2014.09)
Last change: 2015-01-09
(tine20-allinone_2015.07.3.zip)
 
Your logged upgrade plan is of course very useful.

Unfortunately I didn't have time to try the build from Git. But I will definitely make up for that in the near future.

But what I still don't understand at all is; How or where can I select a version?

In the following commands, versions are not specified anywhere?!?

`git clone https://github.com/tine20/tine20.git
cd tine20/tine20
composer install --prefer-source --no-interaction
./vendor/bin/phing`

Maybe you can give me a tip in the right direction, maybe code snippets?

Thanks in advance
Greetings Alex

**Comment by lab-at-nohl** [comment]
Completely untested, but you can clone the tag from the repo. I have chosen **2016.03.5** here because according to my notes I expect it to update fine (although I have used 2016.03.4). 

```php
git clone --depth 1 --branch 2016.03.5 https://github.com/tine20/tine20.git
cd tine20/tine20
cp /my/original/tine/installation/config.inc.php ./
composer install --prefer-source --no-interaction
./vendor/bin/phing
```

Actually I would try to do `php setup.php --update` without running `./vendor/bin/phing` before. Because it might be sufficient to have the php files downloaded to upgrade the database. But I am not sure. Please let me know. 

EDIT: Instead of git clone it, you can also download old versions like this `https://github.com/tine20/tine20/archive/refs/tags/2016.03.5.zip`. 

EDIT 2: And have a backup of your DB in case something goes south. 

Best
Johannes

**Comment by 0NDevil** [comment]
Hi Johannes, 
have tried your commands but when i try to  "composer install xxxx" i get follow error:

`composer install --prefer-source --no-interaction
Installing dependencies from lock file (including require-dev)
Verifying lock file contents can be installed on current platform.
Package operations: 37 installs, 0 updates, 0 removals
  - Syncing tine20/composerapploader (dev-master b9e480b) into cache
Warning: Accessing git.tine20.org over http which is an insecure protocol.

In Git.php line 471:

  Failed to execute git clone --mirror -- 'http://git.tine20.org/git/composerapploader' '/var/www/vhosts/mydomain.de/.cache
  /composer/vcs/http---git.tine20.org-git-composerapploader/'

  Cloning into bare repository '/var/www/vhosts/mydomain.de/.cache/composer/vcs/http---git.tine20.org-git-composerapploader
  '...
  fatal: repository 'http://git.tine20.org/git/composerapploader/' not found


install [--prefer-source] [--prefer-dist] [--prefer-install PREFER-INSTALL] [--dry-run] [--download-only] [--dev] [--no-suggest] [--no-dev] [--no-autoloader] [--no-progress] [--no-install] [--audit] [--audit-format AUDIT-FORMAT] [-v|vv|vvv|--verbose] [-o|--optimize-autoloader] [-a|--classmap-authoritative] [--apcu-autoloader] [--apcu-autoloader-prefix APCU-AUTOLOADER-PREFIX] [--ignore-platform-req IGNORE-PLATFORM-REQ] [--ignore-platform-reqs] [--] [<packages>...]
`


I have an root Server with Plesk v17.x on a Ubuntu v18.x  (pre installed by Hoster)
Don`t know if this is a error because wrong cache tmp directory or a Plesk problem because special Plesk Apache/php Server Config settings, open basedir restrictions or what ever ?!?

But with theese config.inc.php (Settings) i had a succesfully tine20 2018 Installation running on this system (fresh install, no old Data) 
So the tmp directory cannot be so wrong!?
Tried to change the cache directory but the error Message show the same..

any ideas?

Best
Alex

**Reply by lab-at-nohl** [reply]
Can you try to remove composer.lock like `mv composer.lock composer.lock.bkp` before running composer?

EDIT: If that is not sufficient, change lines 8 to 26 of `composer.json` like

```json
 "repositories": [{
      "type": "git",
      "url": "https://github.com/tine20/composerapploader.git"
    }, {
        "type": "git",
        "url": "https://github.com/tine20/zendframework1.git"
    },{
        "type": "git",
        "url": "https://github.com/tine20/OpenDocument.git"
    }, {
        "type": "git",
        "url": "https://github.com/tine20/timezoneconvert.git"
    }, {
        "type": "git",
        "url": "https://github.com/tine20/syncroton.git"
    }, {
      "type": "git",
      "url": "https://github.com/paulmhh/PHPWord"
    }],
```
The repository moved from git.tine20.org to github...
